src/filetype-plugin.js

```diff
@@ -19,32 +19,53 @@
  * That means that the data will end up in the bundle, but is never in the src/
  * directory.
  *
- * The lookup side of this all lives in 'filetype-lookup.ts'.  It can be used
+ * The lookup side of this all lives in 'filetype-lookup.js'.  It can be used
  * with the data imported from './filetype-data' or it can be used directly
  * with the return value of the `create_filetype_data()` function`.
  */
 
-import type { Plugin, PluginBuild } from 'esbuild';
-import language_map from 'language-map' with { type: 'json' };
-import mime_db from 'mime-db/db.json' with { type: 'json' };
+import { readFileSync } from "node:fs";
+import { createRequire } from "node:module";
 
-import { Category, type FileTypeData } from './filetype-lookup.ts';
+import { Category } from './filetype-lookup.js';
 
-export function create_filetype_data() : FileTypeData {
-    const categories = {
+/**
+ * Roughly equivalent to `import(specifier, { with: { type: "json" } })` but
+ * avoids upsetting eslint too much.
```

Comment:
Out of interest, why wasn't it simpler to import the file as plain text (which demonstrably works, given how much Python we import in cockpit), and just wrap a `JSON.parse()` around it?

Reply:
I think it's the very fact of it being a JSON flle.  In that case there are two possible intended things you could do: import it as text or as (parsed) JSON and the type is required to specify what you want.

Reply:
Yeah, I was thinking of renaming it to `.txt`, but I agree that's also nasty. Either way, as I said, not really worth wasting more time on, thanks!